Source/bmalloc/libpas/src/libpas/pas_probabilistic_guard_malloc_allocator.c

```diff
@@ -301,6 +316,64 @@ void pas_probabilistic_guard_malloc_initialize_pgm_as_enabled(void)
     pas_probabilistic_guard_malloc_counter = 0;
 }
 
+/*
+ * Adding the entries into 'tail' index and removing the items from 'head' (FIFO)
+ */
+bool pas_probabilistic_guard_malloc_pgm_metadata_buffer_full(void)
+{
+    return (pgm_metadata_count >= MAX_PGM_HASH_ENTRIES);
+}
+
+bool pas_probabilistic_guard_malloc_pgm_metadata_buffer_empty(void)
+{
+    return (pgm_metadata_count == 0);
+}
+
+void pas_probabilistic_guard_malloc_pgm_metadata_buffer_add(pas_ptr_hash_map_entry * entry)
+{
+    PAS_ASSERT(!pas_probabilistic_guard_malloc_pgm_metadata_buffer_full());
+
+    size_t index = pgm_metadata_tail_index;
+    if (index >= MAX_PGM_HASH_ENTRIES)
+        index = 0;
+    pgm_metadata_vector[index] = entry;
+    pgm_metadata_tail_index = index + 1;
+    pgm_metadata_count++;
+}
+
+void pas_probabilistic_guard_malloc_pgm_metadata_buffer_remove(void)
+{
+    PAS_ASSERT(!pas_probabilistic_guard_malloc_pgm_metadata_buffer_empty());
+
+    size_t index = pgm_metadata_head_index;
+    if (index >= MAX_PGM_HASH_ENTRIES)
+        index = 0;
+
+    pas_ptr_hash_map_entry *entry = pgm_metadata_vector[index];
+    if (entry && entry->value) {
+        pas_pgm_storage *value = (pas_pgm_storage *) entry->value;
+        bool removed = pas_ptr_hash_map_remove(&pas_pgm_hash_map, (void*)entry->key, NULL, &pas_large_utility_free_heap_allocation_config);
+        PAS_ASSERT(removed);
+        pas_utility_heap_deallocate(value);
+    }
+    pgm_metadata_head_index = index + 1;
+    pgm_metadata_count--;
+}
+
+void pas_probabilistic_guard_malloc_manage_metadata(pas_ptr_hash_map_entry * entry)
+{
+    /*
+     * Check if PGM metadata circular buffer is full if so free first metadata entry per "FIFO".
+     * Note the length of metadata liveness will depend on how long it takes to fill all 10 entries.
+     */
+    if (pas_probabilistic_guard_malloc_pgm_metadata_buffer_full()) {
+        pas_probabilistic_guard_malloc_pgm_metadata_buffer_remove();
+        pas_probabilistic_guard_malloc_pgm_metadata_buffer_add(entry);
```

Comment:
We do not need the else statement; we can simplify it to

```
if (full())
    remove();

add();
```

Reply:
Fixed